ddtrace/profiling/collector/_memalloc_tb.c

```diff
@@ -195,14 +228,21 @@ traceback_to_tuple(traceback_t* tb)
         PyTuple_SET_ITEM(frame_tuple, 3, empty_string);
         Py_INCREF(empty_string);
 
+        // Set the class
+#if PY_VERSION_HEX >= 0x03090000
+        Py_SET_TYPE(frame_tuple, (PyTypeObject*)ddframe_class);
```

Comment:
should we check that `ddframe_class` is not `NULL` on entry here for sanity?

Reply:
I'm not 100% sure how we might arrive to such a state, but there's no harm in doing it, so I added the check.  The check converts a potential segfault into an exception, so there's still an error we could probably catch.

I suppose it could occur from either of the two memalloc interfaces, since they both call this `traceback_to_tuple` in one way or another.  I'll take a look at that.

Reply:
Yep with the race conditions of the likes of `the memalloc module is already started` I thought it might be better to be safe than sorry here.